commands/operator-sdk/cmd/add/api.go

```diff
@@ -162,7 +165,54 @@ func apiRun(cmd *cobra.Command, args []string) {
 		log.Fatalf("failed to create %v: %v", filePath, err)
 	}
 
-	// TODO: append rbac rule to deploy/rbac/role.yaml
+	// append rbac rule to deploy/role.yaml
+	roleFilePath := filepath.Join(fullProjectPath, "deploy", "role.yaml")
+	roleYAML, err := ioutil.ReadFile(roleFilePath)
+	if err != nil {
+		log.Fatalf("failed to read role manifest %v: %v", roleFilePath, err)
+	}
+	obj, _, err := cgoscheme.Codecs.UniversalDeserializer().Decode(roleYAML, nil, nil)
+	if err != nil {
+		log.Fatalf("failed to decode role manifest %v: %v", roleFilePath, err)
+	}
+	switch role := obj.(type) {
+	// TODO: use rbac/v1.
+	case *rbacv1beta1.Role:
```

Comment:
Lets just change this to rbacv1. The change is small enough.
We just need to update the scaffold templates for the role and rolebinding as well.
https://github.com/operator-framework/operator-sdk/blob/9465a954670cd28fe343c8e73dafbc8d889686a3/pkg/scaffold/rolebinding.go#L47
https://github.com/operator-framework/operator-sdk/blob/9465a954670cd28fe343c8e73dafbc8d889686a3/pkg/scaffold/role.go#L47

Reply:
The UniversalDeserializer doesn't decode the yaml into `rbacv1.Role` but to `rbacv1beta1.Role`.  That's why i use v1beta1 instead. I suspect there is something with our k8s version being `1.10.1`.

Reply:
Can we do that in a separate pr. The change requires some debugging.

Reply:
Sure, let's do that in a follow up PR along with updating the deps to `1.11.2`

Reply:
yep